src/components/Post.tsx

```diff
@@ -77,6 +77,13 @@ const components = {
     target = isAnchor ? '_blank' : target
     rel = isAnchor ? 'noopener noreferrer' : rel
 
+    const [path, hash] = href.split('#')
+    const isMarkdown = path.endsWith('.md')
+
+    if(!isAnchor && isMarkdown) {
+      href = path.slice(0, -3) + (hash ? `#${hash}` : '')
+    }
+
```

Comment:
If this is just concerned with relative paths that would otherwise work on GitHub, maybe we can focus this case on markdown extensions.

```js
// You'd have to export this from utils/docs.ts and remove `$` from the previous regex
export const MARKDOWN_REGEX = /\.mdx?/

// Post.tsx
import { MARKDOWN_REGEX } from 'utils/docs'

if (!isAnchor) href = href.replace(MARKDOWN_REGEX, '')
```

Reply:
Thanks Cody. I have updated the PR to include the existing (slightly modified) regex.

I am trying to track down what I think is an existing issue. Before approving, can you confirm this is existing: 

> Hydration failed because the initial UI does not match what was rendered on the server.
> Warning: Prop `className` did not match. Server: "rounded-md block px-6 py-2 text-gray-800 font-normal hover:bg-gray-50 cursor-pointer bg-gray-100" Client: "rounded-md block px-6 py-2 text-gray-800 font-normal hover:bg-gray-50 cursor-pointer"

I suspect it's because the active prop in NavItem is different depending if its server rendered or not? 
If its existing, I'm happy to raise as a separate PR

Reply:
Yes, this is an existing issue. Good find though, a PR for that would be appreciated.